### PR TITLE
doc: Added release notes for 3.10.0

### LINKED ISSUE #206: Update release-notes section with release-3.10 notes as well
Filing this issue in advance of 3.10 release, so that once the release announcements are made, the release notes from [1] can be posted on the doc site as well.

[1] Link will work once 3.10.0 is tagged: https://github.com/gluster/glusterfs/blob/v3.10.0/doc/release-notes/3.10.0.md

### PR DESCRIPTION
**Pull request by ShyamsundarR**
Added release notes and appropriate updates to index and
mkdocs yml for the same

Fixes #206 

Signed-off-by: Shyam <srangana@redhat.com>

### CODE REVIEW

File: release-notes/3.10.0.md [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+increases in a volume, though the file/directory numbers remain unchanged.
+With this feature, the performance of directory listing is made mostly
+independent of the number of nodes/bricks in the volume. Thus scale doesn't
+exponentially reduce the directory listing performance. (On a 2, 5, 10, 25 brick
+setup we saw ~5, 100, 400, 450% improvement consecutively)
+
+To enable this feature:
+```bash
+# gluster volume set <VOLNAME> performance.readdir-ahead on
+# gluster volume set <VOLNAME> performance.parallel-readdir on
```
Comment (prashanthpai) [inline release-notes/3.10.0.md]: Given the sheer improvement possible, I'm surprised this isn't turned on by default.

File: release-notes/3.10.0.md [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+
+*Limitations:*
+
+*Known Issues:*
+
+### md-cache can optionally -ve cache security.ima xattr
+*Notes for users:*
+From kernel version 3.X or greater, creating of a file results in removexattr
+call on security.ima xattr. This xattr is not set on the file unless IMA
+feature is active. With this patch, removxattr call returns ENODATA if it is
```
Comment (prashanthpai) [inline release-notes/3.10.0.md]: nite: removxattr

**Comment by humblec** [comment]
@prashanthpai sorry abt it. I thought, its an important patch to go in asap as a preparation to the release.  May be I can send one PR on top of this commit.

**Comment by ShyamsundarR** [comment]
@humblec are you sending the PR to correct the above? Otherwise let me know and I can do the same

**Pull request merged**